File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:

      map<int , int>memo;

    int countWays( int n) {


        if(n==0) return 1;


        if(n<0) return 0;



     if(memo[n]) return memo[n];
        


        memo[n] =   countWays(n-1) + countWays(n-2);





      
        return memo[n];

    }



    int climbStairs(int n) {


        if(n<=3) {
            return n;
        }


       





       return countWays(n);

     






    }
       
    
};

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */



function countWays( n , map) {

        if(n===0 || n===1) return 1;

         
        if(map[n]) return map[n];


         

         map[n]= countWays(n-1 , map)  + countWays(n-2 , map);


         return map[n];


}
var climbStairs = function(n) {

    const map = {}

    return countWays( n , map)
    
};